platform-channels.md

```diff
@@ -0,0 +1,249 @@
+[PlatformMethodChannel]: https://docs.flutter.io/flutter/services/PlatformMethodChannel-class.html
+[FlutterMessageChannelAndroid]: https://docs.flutter.io/javadoc/io/flutter/plugin/common/FlutterMessageChannel.html
+[FlutterMethodChannelAndroid]: https://docs.flutter.io/javadoc/io/flutter/plugin/common/FlutterMethodChannel.html
+[FlutterChanneliOS]: https://github.com/flutter/engine/blob/master/shell/platform/darwin/ios/framework/Source/FlutterChannels.mm
+
+### Platform channel data types support and codecs
+
+The standard platform channels use a standard message codec that supports
+efficient binary serialization of simple JSON-like values, such as booleans,
+numbers, Strings, and List and Maps of these (see
+[`StandardMessageCodec`](https://docs.flutter.io/flutter/services/StandardMessageCodec-class.html))
+for details). The serialization and deserialization of these values to and from
+messages happens automatically when you send and receive values.
+
+For more specialized use cases, you may also use the
+[`BinaryCodec`](https://docs.flutter.io/flutter/services/BinaryCodec-class.html),
+[`StringCodec`](https://docs.flutter.io/flutter/services/StringCodec-class.html),
+or
+[`JSONMessageCodec`](https://docs.flutter.io/flutter/services/JSONMessageCodec-class.html).
+
+### Example: Calling platform-specific iOS and Android code using platform channels
+
+The following example demonstrates how to call a new platform-specific API,
+through a concrete example that retrieves and displays the current battery level
+by calling into the Android `BatteryManager` API, and the iOS
+`device.batteryLevel` API, via a single platform message, `getBatteryLevel`.
+
+The standard app template created when you create a new Flutter app can be used.
+
+*Note*: The full source-code for this example is available in
+[`/samples/platform-services/`](https://github.com/flutter/flutter/tree/master/examples/platform_services).
+
+**Step 1: Create the Flutter platform client**
+
+We start by adding support for updating our app's `State` class with the current
+battery state.
+
+First, we construct the channel. Since we need just a single platform message,
+`getBatteryLevel`, representing a platform method call, we will use a
+`PlatformMethodChannel`.
+
+The client and host sides of a channel are connected through a channel name
+passed in the channel constructor. In this example we are using `battery`. Note
+that all channel names used in a single app must be unique; we recommend using a
+naming pattern *TODO*.
+
+<!-- skip -->
+```dart
+class _PlatformServicesState extends State<PlatformServices> {
+  // Construct platform channel inside State class.
+  static const platform = const PlatformMethodChannel('battery');
+
+  // TODO: Get battery level.
+}
+```
+
+Next, we invoke a method on the method channel, specifying the concrete method
+to call via a String `method` argument. We then use the returned result to update
```

Comment:
Please reconsider the word `argument`, as it may be confusing: there is also the question of passing arguments to the method called on the channel. That is not shown in your example, but should probably be mentioned here.